### PR TITLE
docs(best-practices): conventional commit branch names

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Add section that explains Conventional Commit branch names

## Context

Good branch names make it easier for the maintainers to track what each branch is supposed to do.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/development/best-practices.md [hunk 1]
```diff
@@ -3,6 +3,26 @@
 This document explains our best practices.
 Follow these best practices when you're working on our code.
 
+## Git branch names
```
Comment (HonkingGoose) [inline docs/development/best-practices.md:6]: I think it's a good idea to put this section near the top, as that way it's the first thing you see before you start creating branches. But if you want I can move it someplace else.

File: docs/development/best-practices.md [hunk 2]
```diff
@@ -3,6 +3,26 @@
 This document explains our best practices.
```
Comment (HonkingGoose) [inline docs/development/best-practices.md:3]: I thought of putting the branch name stuff in the `local-development.md` and `remote-development.md` docs. But then I would have the same information in two files.

So I'm putting the branch name stuff in `best-practices.md` instead. 😄

File: docs/development/best-practices.md [hunk 3]
```diff
@@ -3,6 +3,26 @@
...
 Follow these best practices when you're working on our code.
 
+## Git branch names
+
+The Renovate maintainers checkout a lot of Git branches.
+When Git branches have names like `patch-1` it's hard to remember what each branch is about.
+You can help the maintainers by following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification when naming your Git branch.
```
Comment (rarkins) [inline docs/development/best-practices.md:10]: Do you think we can remove this, and get to the point?

File: docs/development/best-practices.md [hunk 4]
```diff
@@ -3,6 +3,26 @@
...
+
+Branch names should start with a Conventional Commit scope like `feat/` or `fix/` for example:
+
+- `feat/13732-cacache-cleanup`
```
Comment (rarkins) [inline docs/development/best-practices.md:14]: Now I think about it, putting the issue number like above is important too, so we should explicitly mention it

File: docs/development/best-practices.md (hunk 3 above)
Comment (HonkingGoose) [reply docs/development/best-practices.md:10]:
```suggestion
```

Yeah, let's remove the introduction and fast-forward to the instructions. I'm putting the link to the Conventional Commits spec to another sentence.

File: docs/development/best-practices.md [hunk 5]
```diff
@@ -3,6 +3,26 @@
 This document explains our best practices.
 Follow these best practices when you're working on our code.
 
+## Git branch names
+
+The Renovate maintainers checkout a lot of Git branches.
+When Git branches have names like `patch-1` it's hard to remember what each branch is about.
+You can help the maintainers by following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification when naming your Git branch.
+
+Branch names should start with a Conventional Commit scope like `feat/` or `fix/` for example:
```
Comment (HonkingGoose) [inline docs/development/best-practices.md:12]:
```suggestion
Branch names must start with a [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) scope like `feat/` or `fix/`.
If you're closing an issue with your PR: mention the GitHub issue number.

For example:
```

File: docs/development/best-practices.md [hunk 6]
```diff
@@ -3,6 +3,26 @@
...
+## Git branch names
+
+
+Branch names must start with a [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) scope like `feat/` or `fix/`.
+If you're closing an issue with your PR then put the issue number after that.
+
+For example:
+
+- `feat/13732-cacache-cleanup`
+- `fix/gitea-automerge-strategy`
```
Comment (rarkins) [inline docs/development/best-practices.md:15]:
```suggestion
- `fix/15431-gitea-automerge-strategy`
```

**Review by rarkins — DISMISSED** [review]

**Pull request merged**